### DISCUSSION TITLE
Agent relay through agent?

### DISCUSSION THREAD

**Discussion by norbythegeek** [Ylianst/MeshCentral#5278] [General]
First, I just found MeshCentral, and love it.  This will enable my small cloud-based company to ditch our MSP that I'm currently only really using for remote access.  MeshCentral is much lighter than SolarWinds TakeControl and much less expensive.

I have a unique use case.  We run a testing lab for ISP equipment, and have a VM connected to the user-end network equipment.  The VM's default route goes through the test path, so we can run throughput tests.  The VM also has another "back-door" interface that connects to the local lab LAN for remote access.

Since the VM only has Internet access if the test path is up and running, I currently have an agent loaded on an always-connected machine on the main LAN, and relay RDP through there to the test VMs.  Would it be possible to install an agent on the test VMs, but still have their connection route through the relay?  This would give me the flexibility of running a full agent on the test VMs, and still have them always-connected to the MeshCentral server.

Thanks!

**Comment by si458** [comment]
how is ur meshcentral hosted? 
if meshcentral is in the cloud on the internet then the VM would need to connect back to it via ur test path,
BUT 
if the VM is attached to the same network as meshcentral, then you could use the LAN option and i believe you can still connect to it IF your testpath does down

but as a general rule the meshagent will always use the DEFAULT ROUTE (in ur case the testpath) to connect to your meshcentral server, unless it can be access via another way

ie: mesh ip 192.168.88.99 -> vm ip 192.168.89.99 but the VM has a route to 192.168.88.99 WITHOUT going to the gateway first using iptables or windows route table

**Reply by norbythegeek** [reply]
Our MeshCentral server is in the cloud.

I think you're suggesting I create a static route in the test VM for the MeshCentral Cloud IP to go back out through the LAN interface, I'll try that.  I'm not sure why I didn't think of that already.  Thanks!

**Reply by si458** [reply]
to be honest i didnt even think of that myself haha! but yeh i think that would work! again give it a try 👍

**Reply by norbythegeek** [reply]
It does work, I just have to make sure the test VM also gets DNS resolution from the LAN as well.

In an admin command prompt:
route -p [meshcentral IP] mask 255.255.255.255 [LAN Gateway IP]
